selene_sdk/predict/_in_silico_mutagenesis.py

```diff
@@ -39,7 +48,36 @@ def in_silico_mutagenesis_sequences(sequence,
         we return a list with length of 3000-4000, depending on the number of
         unknown bases in the input sequences.
 
+    Raises
+    ------
+    ValueError
+        If the value of `start_position` or `end_position` is negative.
+    ValueError
+        If there are fewer than `mutate_n_bases` between `start_position`
+        and `end_position`.
+    ValueError
+        If `start_position` is greater or equal to `end_position`.
+    ValueError
+        If `start_position` is not less than `len(sequence)`.
+    ValueError
+        If `end_position` is greater than `len(sequence)`.
+
     """
+    if end_position is None:
+        end_position = len(sequence)
+    if start_position >= end_position:
+        raise ValueError("Starting positions must be less than the ending positions.")
+    if start_position < 0:
+        raise ValueError("Negative starting positions are not supported.")
+    if end_position < 0:
+        raise ValueError("Negative ending positions are not supported.")
+    if start_position >= len(sequence):
+        raise ValueError("Starting positions must be less than the sequence length.")
```

Comment:
For any errors that mention sequence length, can you also print the value of `len(sequence)` the expected sequence length? Helpful for people using the CLI who might not be sure of what the expected sequence length is.

Reply:
This has been updated in the recent commit. All errors related to selective ISM will print relevant values.